linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/HttpConfig.scala

```diff
@@ -52,13 +52,27 @@ class HttpInitializer extends ProtocolInitializer.Simple {
 
 object HttpInitializer extends HttpInitializer
 
+case class HttpClientConfig(
+  engine: Option[HttpEngine]
+) extends ClientConfig {
+  override def clientParams =
+    engine.foldLeft(super.clientParams) { (params, engine) => engine.mk(params) }
```

Comment:
TIOLI: it's probably just me, but for some reason I get confused when folding over options.  
```
engine match {
  case Some(eng) => eng.mk(super.clientParams)
  case None => super.clientParams
}```

is maybe more obvious? FFTI